contributors/design-proposals/storage/csi-migration.md

```diff
@@ -321,6 +323,88 @@ with minimal disruption to users would be:
 And at this point users doing their own deployment and not installing the GCE PD
 CSI driver encounter an error.
 
+## Plugin Availability (Installation)
+When Migration goes Beta and is turned on by default we have the issue of how to
+get the drivers on to the clusters. We have to consider what to do with both new
+clusters, as well as existing clusters that may already be using storage. We
+also have to consider what to do with both cloud storage solutions, and all the
+other non-cloud specific drivers.
+
+### New Clusters (Cloud Drivers)
+For new clusters we can modify the `cluster-up` scripts to deploy the cloud
+storage drivers automatically with cluster bring-up depending on which cloud the
+cluster is being brought up on. For example if we `cluster-up` on GCE we would
+just install the GCE PD CSI Driver by default. Each Cloud Provider can also
+provide their own mechanism adding drivers in their own managed solutions.
+
+### Upgraded Clusters (Cloud Drivers)
+For upgraded clusters the respective cloud drivers should be installed on
+upgrade. If there is no good universal hook to install things onto a cluster on
+upgrade then we fallback to the behavior described in the next section "Missing
+Drivers."
+
+### Non-Cloud and Missing Drivers
+For non-cloud and other miscellaneous missing drivers the question of
+installation becomes more nuanced. We have to strike a balance between
+unexpected resource consumption (installing everything would be unnecessary and
+wasteful) and user grief (in-tree drivers no longer work because CSI Drivers are
+not installed). There are a couple options we could take here:
+
+#### Automatic Installation
+One option is to create a tool that installs CSI Drivers automatically based on
+storage classes present. Since the storage classes for in-tree plugins have
```

Comment:
Storage classes are optional, scanning storage classes *and* PVs is more reliable.